packages/runtime/test/transport/relationshipTemplates.test.ts

```diff
@@ -247,6 +248,173 @@ describe("RelationshipTemplate Tests", () => {
+                    content: emptyRelationshipTemplateContent,
+                    expiresAt: DateTime.utc().plus({ minutes: 1 }).toString(),
+                    passwordProtection: {
+                        password: "password"
+                    }
+                })
+            ).value.id;
+            const createResult = await runtimeServices1.transport.relationshipTemplates.createTokenForOwnTemplate({ templateId });
+
+            const loadResult = await runtimeServices2.transport.relationshipTemplates.loadPeerRelationshipTemplate({
+                reference: createResult.value.truncatedReference,
+                password: "password"
+            });
+            expect(loadResult).toBeSuccessful();
+            expect(loadResult.value.passwordProtection!.password).toBe("password");
+            expect(loadResult.value.passwordProtection!.passwordIsPin).toBeUndefined();
+        });
+
+        test("send and receive a PIN-protected template via a token", async () => {
+            const templateId = (
+                await runtimeServices1.transport.relationshipTemplates.createOwnRelationshipTemplate({
+                    content: emptyRelationshipTemplateContent,
+                    expiresAt: DateTime.utc().plus({ minutes: 1 }).toString(),
+                    passwordProtection: {
+                        password: "1234",
+                        passwordIsPin: true
+                    }
+                })
+            ).value.id;
+
+            const createResult = await runtimeServices1.transport.relationshipTemplates.createTokenForOwnTemplate({ templateId });
+
+            const loadResult = await runtimeServices2.transport.relationshipTemplates.loadPeerRelationshipTemplate({
+                reference: createResult.value.truncatedReference,
+                password: "1234"
+            });
+            expect(loadResult).toBeSuccessful();
+            expect(loadResult.value.passwordProtection!.password).toBe("1234");
+            expect(loadResult.value.passwordProtection!.passwordIsPin).toBe(true);
+        });
+
+        test("error when loading a template with a wrong password", async () => {
+            const createResult = await runtimeServices1.transport.relationshipTemplates.createOwnRelationshipTemplate({
+                content: emptyRelationshipTemplateContent,
+                expiresAt: DateTime.utc().plus({ minutes: 1 }).toString(),
+                passwordProtection: {
+                    password: "password"
+                }
+            });
+            expect(createResult).toBeSuccessful();
+
+            const loadResult = await runtimeServices2.transport.relationshipTemplates.loadPeerRelationshipTemplate({
+                reference: createResult.value.truncatedReference,
+                password: "wrong-password"
+            });
+            expect(loadResult).toBeAnError(/.*/, "error.runtime.recordNotFound");
```

Comment:
If the password is wrong, we'll throw a `recordNotFound` error with the message `'RelationshipTemplate not found. Make sure the ID exists and the record is not expired.'`. We should be more precise with the error, such that someone doesn't think the RelationshipTemplate is the problem if they provide a wrong password. If no password is provided, we'll throw an error `error.transport.noPasswordProvided` with message `"You need to provide a password to perform this operation."`. That means, we already expose the information that the RelationshipTemplate requires a password. Thus, I suggest to add an error that informs about a wrong password.

Reply:
We can not do this. The backbone doesn't give more information..

Reply:
What about expanding the error message to also include a failed password check as a cause?

Reply:
Hmmmmmm, forget about my comment. 

1. we know that the thing is pw protected
2. we get a 404 from the bkb
3. we COULD say that there is a possibility that
  a. the thing is expired
  b. the password is incorrect

But we should tell about both issues.

Reply:
> What about expanding the error message to also include a failed password check as a potential cause?

Yeah, but please only expand if we think that is's password protected.

Reply:
> We can not do this. The backbone doesn't give more information..

Should we request this from the Backbone then? It would be good to be able to differentiate these cases.

> What about expanding the error message to also include a failed password check as a potential cause?

That would be an alternative, if the Backbone has valid reasons not to give us more information. We could in fact only expand the error message if we know that the reference has a passwordProtection.

Reply:
And - please check the message instead of doing `/.*/` then.

Reply:
> Should we request this from the Backbone then? It would be good to be able to differentiate these cases.
> 
> >

It's a FEATURE, that we (and noone else) don't get the information.

Reply:
I thought the Backbone blocks the RelationshipTemplate after a certain amount of failed attempts to avoid bruteforce.

Reply:
Not yet. But this would only be an additional security mechanism. It's an antipattern to say a password is wrong.

Reply:
Alright, then let's expand the error message in case we have a passwordProtection and get an error from the Backbone. 👍

Reply:
alrighty schmighty!